File: Leaders in an array/Leaders in an array.cpp
class Solution{
    //Function to find the leaders in the array.
    public:
    vector<int> leaders(int a[], int n){
    vector<int> ans;
        int left=0,right=0;
        while(left<n)
        {
            if(a[left]>=a[right+1])
            {
                right++;
                if(right==n-1){
                   ans.push_back(a[left]);
                   left++;
                   right=left;
                }
            }
            else{
                left++;
                right=left;
            }
        }
        ans.push_back(a[n-1]);
        return ans;
    }
};

File: Leaders in an array/Leaders in an array.py
class Solution:
    #Back-end complete function Template for Python 3
    
    #Function to find the leaders in the array.
    def leaders(self, A, N):
        #Code here
        max = A[N-1]
        result = [max]
        for i in range(N-2,-1,-1):
            if A[i] >= max:
                max = A[i]
                result.append(max)
        result.reverse()
        return result